pkg/controller/nodes/executor.go

```diff
@@ -429,14 +554,19 @@ func (c *nodeExecutor) handleQueuedOrRunningNode(ctx context.Context, nCtx *node
 		np = v1alpha1.NodePhaseSucceeded
 		finalStatus = executors.NodeStatusSuccess
 	}
+	if np == v1alpha1.NodePhaseRecovering && !h.FinalizeRequired() {
```

Comment:
if it is recovering we do not even care about the  handler or finalization right?

Reply:
do we really need `Recovering`?

Reply:
updated